esm_plugin_manager/esm_plugin_manager.py

```diff
@@ -120,10 +120,18 @@ def work_through_recipe(recipe, plugins, config):
         import pdb
 
         pdb.set_trace()
-    for workitem in recipe["recipe"]:
+    recipes = recipe["recipe"]
+    for index, workitem in enumerate(recipes):
         if config["general"].get("verbose",False):
-            print(workitem)
-            print("-" * len(workitem))
+            # diagnostic message of which recipe step is being executed
+            message = (f'::: Executing the step:  {workitem}    '
+            f'(step [{index+1}/{len(recipes)}] of the job:  '
```

Comment:
I think you can remove the `+1` in `index+1` `f'(step [{index+1}/{len(recipes)}] of the job:  '` if you write the `enumerate` like this:
```
for index, workitem in enumerate(recipes, start=1):
```

Reply:
Thanks for the comment. One side effect of programming for a long time is you may find yourself start counting from 0. 😆